### DISCUSSION TITLE
support for Windows 10 Pro

### DISCUSSION THREAD

**Discussion by VesnaVojinovic** [sctplab/usrsctp#648] [Q&A] [answered]
"The user-land stack has been tested on FreeBSD 10.0, OpenBSD 7.0, Ubuntu 11.10, Windows 7, "
Does it support Windows 10 Pro ?

I tried "nmake -f Makefile.nmake" on my Windows 10 Pro and nmake from Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.28.29333\bin\Hostx64\x86\nmake.EXE but it fails with
.\user_environment.h(34): fatal error C1083: Cannot open include file: 'sys/types.h': No such file or directory
Am I doing something wrong ?

**Comment by tuexen** [comment]
I haven't used the `nmake` tool for a while, actually don't have access to a Windows system right now. I don't see a reason, why the library can't work on Windows 10 Pro. Maybe some tweaks for building are required. Does the library build when you use `cmake` instead of `nmake`?

**Reply by VesnaVojinovic** [reply]
I was OFF last two days, so the answer now:

cmake generally works but building Win32 version (cmake -A Win32) I got the error during compiling of rtcweb.c :
D:\Projects\WebRTC\usrsctp\programs\rtcweb.c(933,7): error C2220: the following warning is treated as an error [D:\Projects\WebRTC\win32_fails\programs\rtcwe
b.vcxproj]
D:\Projects\WebRTC\usrsctp\programs\rtcweb.c(933,7): warning C4996: '_snprintf': This function or variable may be unsafe. Consider using _snprintf_s instead.
 To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details. [D:\Projects\WebRTC\win32_fails\programs\rtcweb.vcxproj]
 
 Using "stand-alone" 
 #ifdef _WIN32
#define _CRT_SECURE_NO_WARNINGS
#endif
at the very beginning of rtcweb.c solves the problem.

**Reply by tuexen** [reply]
Yeah, I'm aware of them, will fix them. So cmake works, but nmake does not.

**Reply by VesnaVojinovic** [reply]
to be honest, I have not tried nmake again since I got usrsctp built

**Reply by VesnaVojinovic** [reply]
when I have more time, I'll try it

**Reply by VesnaVojinovic** [reply]
regarding nmake - it works fine
That was my "error" - I had to reinstall/repair my VS 2019

**Comment by tuexen** [comment, marked as answer]
Thanks for letting me know. @msvoelker did some testing and we figured out that installing a Windows SDK was necessary. I also fixed the warnings related to `rtcweb.c`.
